robottelo/host_helpers/satellite_mixins.py

```diff
@@ -351,3 +352,8 @@ def api_factory(self):
         if not getattr(self, '_api_factory', None):
             self._api_factory = APIFactory(self)
         return self._api_factory
+
+    def ui_factory(self, session):
+        if not getattr(self, '_ui_factory', None):
+            self._ui_factory = UIFactory(self, session=session)
+        return self._ui_factory
```

Comment:
I think the current version is a holdover from this being treated more like a property. Now that we're requiring a session being passed each time, turning it into a cached method would help.
The only thing I'm current unsure of is if the Session objects are hashable. If not, I can help with that.
```suggestion
    @cache
    def ui_factory(self, session):
        return UIFactory(self, session=session)
```